packages/react/src/hooks/useStorageURL.ts

```diff
@@ -0,0 +1,32 @@
+import { S3ProviderGetConfig, Storage } from '@aws-amplify/storage';
+import { useEffect, useState } from 'react';
+
+export interface UseStorageURLResult {
+  url?: string;
+  error?: Error;
+  isLoading: boolean;
+}
+
+export const useStorageURL = (key: string, options?: S3ProviderGetConfig) => {
+  const [result, setResult] = useState<UseStorageURLResult>({
+    isLoading: true,
+  });
+
+  const fetch = () => {
+    setResult({ isLoading: true });
+
+    const promise = Storage.get(key, options);
+
+    // Attempt to fetch storage object url
+    promise
+      .then((url) => setResult({ url, isLoading: false }))
+      .catch((error) => setResult({ error, isLoading: false }));
+
+    // Cancel current promise on unmount
+    return () => Storage.cancel(promise);
+  };
+
+  useEffect(fetch, []);
```

Comment:
I think you should either:
1. Wrap the `fetch` function in a `useCallback` or
2. Inline the function in the `useEffect` hook itself

Currently the dependencies are not appropriately addressed. The `key` and `options` props are supposed to go into the dependency array of the `useEffect` hook since you would want to get a new URL when the `key` or `options` change.

Reply:
good call! added `key` and a serialized version of `options` as part of the dependency list